lms/services/h_api.py

```diff
@@ -196,6 +196,9 @@ def get_annotation_counts(
         h_userids: list[str] | None = None,
         resource_link_ids: list[str] | None = None,
     ) -> list[AnnotationCounts]:
+        if not group_authority_ids:
```

Comment:
If we found not groups to query H the results is that none will have annotations. This is probably not common in production but I seen it happen locally.

When calling H with groups we get an exceptions back so this both fixes a bug and saves a HTTP request.